rc/tools/clipboard.kak

```diff
@@ -0,0 +1,94 @@
+declare-option -docstring %{
+    The shell command to which data will be piped to copy to the system clipboard
+} str clipboardcmd_copy %sh{
+    for cmd in "xsel -i" "pbcopy" "wl-copy -p"; do
```

Comment:
You might also want to check for [vis-clipboard](https://github.com/martanne/vis/blob/v0.6/vis-clipboard), a tool from the `vis` editor that abstracts over all those other tools. It's *probably* not going to be installed anywhere Kakoune is, but if it *is* installed, it's pretty nice.

Reply:
I didn't know about this, I'll probably set the `clipboardcmd_*` options to it in my configuration!

I'm not sure this should be in the built-in list though, I envisioned that it would only contain tools that are well-known and ubiquitous.

A similar solution was discussed for `terminalcmd` and `x-terminal-emulator` in #1555.

Reply:
whats the point of wrapping a wrapper that does the same thing?

Reply:
Only have the right tool installed on a given system (or in a container), and not worry whether it's supported by the editor.

Reply:
> whats the point of wrapping a wrapper that does the same thing?

For one, `vis-clipboard` wraps more backends than just xsel, pbcopy and wl-copy. For two, it's smarter about detecting which one to use - for example, it only tries xsel if the X11 `$DISPLAY` variable is set. That's all logic users don't have to re-implement in Kakoune.

Reply:
> for example, it only tries xsel if the X11 $DISPLAY variable is set.

We should probably also do that. Many users have both `xsel` and `wl-{copy,paste}`.
We should first check if `WAYLAND_DISPLAY` is nonempty.

As the logic gets more refined, we should probably use just one shell scope to initialize both `clipboardcmd_copy` and `clipboardcmd_paste`.

Reply:
I don't think verifying whether the tool selected can actually work is necessary, we don't want to re-implement a robust abstraction in what I see as a convenience.

It would make the code heavier, and the editor can reasonably expect an X-less environment not to have `xsel` installed in the first place. I'd leave all that logic to a wrapper like `vis-clipboard` if it's important to the user.

Reply:
I think the additional check is worth it. Consider `nvim` and `pass` who both support copy-to-clipboard on wayland out of the box, even when xsel/xclip are installed.

>  the editor can reasonably expect an X-less environment not to have xsel installed

I think it's also not unreasonable to have both installed.